Demo/Common/IntegrationExamples/Components/ApplePayComponentAdvancedFlowExample.swift

```diff
@@ -0,0 +1,192 @@
+        config.requiredBillingContactFields = [.postalAddress]
+        config.shippingMethods = ConfigurationConstants.shippingMethods
+
+        let component = try? ApplePayComponent(paymentMethod: paymentMethod,
+                                               context: context,
+                                               configuration: config)
+        component?.delegate = self
+        return component
+    }
+
+    // MARK: - Payment response handling
+
+    private func paymentResponseHandler(result: Result<PaymentsResponse, Error>) {
+        switch result {
+        case let .success(response):
+            finish(with: response)
+        case let .failure(error):
+            finish(with: error)
+        }
+    }
+
+    internal func finish(with result: PaymentsResponse) {
+        let success = result.resultCode == .authorised || result.resultCode == .received || result.resultCode == .pending
+        let message = "\(result.resultCode.rawValue) \(result.amount?.formatted ?? "")"
+        finalize(success, message)
+    }
+
+    internal func finish(with error: Error) {
+        let message: String
+        if let componentError = (error as? ComponentError), componentError == ComponentError.cancelled {
+            message = "Cancelled"
+        } else {
+            message = error.localizedDescription
+        }
+        finalize(false, message)
+    }
+
+    private func finalize(_ success: Bool, _ message: String) {
+        applePayComponent?.finalizeIfNeeded(with: success) { [weak self] in
+            guard let self = self else { return }
+            self.dismissAndShowAlert(success, message)
+        }
+    }
+
+    internal func dismissAndShowAlert(_ success: Bool, _ message: String) {
+        presenter?.dismiss {
+            // Payment is processed. Add your code here.
+            let title = success ? "Success" : "Error"
+            self.presenter?.presentAlert(withTitle: title, message: message)
+        }
+    }
+
+    private func presentAlert(with error: Error, retryHandler: (() -> Void)? = nil) {
+        presenter?.presentAlert(with: error, retryHandler: retryHandler)
+    }
+
+}
+
+extension ApplePayComponentAdvancedFlowExample: PaymentComponentDelegate {
+
+    internal func didSubmit(_ data: PaymentComponentData, from component: PaymentComponent) {
+        let request = PaymentsRequest(data: data)
+        apiClient.perform(request) { [weak self] result in
+            self?.paymentResponseHandler(result: result)
+        }
+    }
+
+    internal func didFail(with error: Error, from component: PaymentComponent) {
+        finish(with: error)
+    }
+
+}
+
+extension ApplePayComponentAdvancedFlowExample: PresentationDelegate {
+
+    internal func present(component: PresentableComponent) {
+        let componentViewController = component.viewController
+        presenter?.present(viewController: componentViewController, completion: nil)
+    }
+}
```

Comment:
```suggestion
```

This is not needed for advanced flow? There is no sessions and there is no ActionComponent?

Reply:
yes, you are right :) 434daaef01d20f3f62437a7110b7baf93ccfb13f